eva_cttv_pipeline/trait_mapping/trait_names_parsing.py

```diff
@@ -32,23 +32,19 @@ def get_trait_names(clinvar_json: dict) -> list:
         trait_set = clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["traitSet"]
     else:
         trait_set = clinvar_json["referenceClinVarAssertion"]["traitSet"]
-    trait_list = []
+    trait_names_to_return = []
     for trait in trait_set['trait']:
-        trait_list.append([])
+        trait_name = None
         for name in trait['name']:
             # First trait name in the list will always be the "Preferred" one
             if name['elementValue']['type'] == 'Preferred':
-                trait_list[-1] = [name['elementValue']['value']] + trait_list[-1]
+                trait_name = name['elementValue']['value']
             elif name['elementValue']['type'] in ["EFO URL", "EFO id", "EFO name"]:
                 continue  # if the trait name not originally from clinvar
-            else:
-                trait_list[-1].append(name['elementValue']['value'])
-
-    trait_names_to_return = []
-    for trait in trait_list:
-        if len(trait) == 0:
-            continue
-        trait_names_to_return.append(trait[0].lower())
+            elif trait_name is None:
+                trait_name = name['elementValue']['value']
```

Comment:
I'm first checking if trait_name is None before I assign this, so this wouldn't happen. I've just pushed another commit where the loop breaks upon a preferred name being found, so it shouldn't continue looping anyway after that now.